Resources/Prototypes/Entities/Structures/Power/Generation/Tesla/energyball.yml

```diff
@@ -0,0 +1,106 @@
+- type: entity
+  id: TeslaEnergyBall
+  name: ball lightning
+  description: A giant ball of pure energy. The space around it is humming and melting.
+  components: 
+  - type: Clickable
+  - type: AmbientSound
+    volume: 2
+    range: 14
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+  - type: CanMoveInAir
+  - type: Fixtures
+    fixtures:
+      TeslaCollider:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.35
+        hard: false
+        restitution: 0.8
+        density: 10
+        mask:
+        - None
+        layer:
+        - AllMask 
+  - type: TeslaEnergyBall
+    energyLoss: 3
+    spawnProto: TeslaMiniEnergyBall 
+  - type: LightningArcShooter
+    shootRange: 10
+    shootMinInterval: 2
+    shootMaxInterval: 5
+    lightningPrototype: HyperchargedLightning
+  - type: ChasingWalk # To make the tesla move around.
```

Comment:
The second problem is that Tesla shouldn't react to explosions. Now, if there is an explosion close to the tesla (and it often explodes things around it), either everything will be fine, or the tesla will be pushed VERY far away, or the tesla will have something broken with physics, and impulses from ChasingWalk will almost stop moving the tesla, in fact, it becomes immobile.

Reply:
Ideally, this could be solved simply by making the tesla immune to the effects of explosions.
But you have to leave the ability to go through walls, but not through the energy barrier.....

Reply:
is singularity immune to explosions?

Reply:
I just checked, the singularity is also very good at repelling explosions, hah.

Reply:
Progress on researching the problem: After being hit by the explosion, thesla's Momentum becomes 0. And Impulses received from ChasingWalk component push a couple of centimeters without changing Momentum,

The Momentum of a freshly created Tesla is equal to the current direction of motion.

Reply:
Normal Movement:

https://github.com/space-wizards/space-station-14/assets/96445749/9326c1e1-8bb3-4f93-a5d1-040e6c68642c

Reply:
Movement after exploded:

https://github.com/space-wizards/space-station-14/assets/96445749/44bdd789-aee8-45ed-bc50-181bdc070b85

Reply:
Looks like the tesla is either getting grounded (physics bodies status set to OnGround) or its becoming a kinematic body for some reason? Probably the former.

Reply:
Thank you! That was exactly what I needed, I fixed that problem.

There's still the problem that sometimes a tesla explosion pushes the tesla very far away